workspaces/arborist/test/arborist/build-ideal-tree.js

```diff
@@ -833,6 +833,46 @@ t.test('workspaces', t => {
     )
   })
 
+  t.test('should allow adding a workspace as a dep to a workspace', async t => {
+    // turn off networking, this should never make a registry request
+    nock.disableNetConnect()
+    t.teardown(() => nock.enableNetConnect())
+
+    const path = t.testdir({
+      'package.json': JSON.stringify({
+        name: 'root',
+        workspaces: ['workspace-a', 'workspace-b'],
+      }),
+      'workspace-a': {
+        'package.json': JSON.stringify({
+          name: 'workspace-a',
+          version: '1.0.0',
+        }),
+      },
+      'workspace-b': {
+        'package.json': JSON.stringify({
+          name: 'workspace-b',
+          version: '1.0.0',
+        }),
+      },
+    })
+
+    const arb = new Arborist({
+      ...OPT,
+      path,
+      workspaces: ['workspace-a'],
+    })
+
+    const tree = await arb.buildIdealTree({
+      path,
+      add: [
+        'workspace-b',
+      ],
+    })
+
+    t.matchSnapshot(printTree(tree))
```

Comment:
I wish this test had assertions to make it more explicit on the behavior you're looking for.

Reply:
It's specifically "the output is tab-indented" - other than ensuring that each line starts with either "not a space" or "a tab", i'm not sure what i'd do. what would you prefer?